packages/uui-modal/lib/uui-modal-sidebar.element.ts

```diff
@@ -0,0 +1,100 @@
+import { css, html, PropertyValueMap } from 'lit';
+import { customElement } from 'lit/decorators.js';
+import { UUIModalElement } from './uui-modal.element';
+
+@customElement('uui-modal-sidebar')
+export class UUIModalSidebarElement extends UUIModalElement {
+  constructor() {
+    super();
+    this.addEventListener('close', this.#onClose.bind(this));
+  }
+
+  protected firstUpdated(
+    _changedProperties: Map<string | number | symbol, unknown>
+  ): void {
+    super.firstUpdated(_changedProperties);
+
+    this.style.setProperty('--uui-modal-offset', -this.#getWidth + 'px');
+  }
+
+  protected updated(
+    _changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>
+  ): void {
+    super.updated(_changedProperties);
+
+    // if we've reached over X layers of sidebars, hide for better performance.
+    // TODO: We could instead hide the sidebars when out of the viewport, but this is a good start.
+    if (this.uniqueIndex > 10) {
+      this.setAttribute('hide', '');
+    } else {
+      this.removeAttribute('hide');
+    }
+  }
+
+  get #getWidth() {
+    return this._dialogElement?.getBoundingClientRect().width ?? 0;
+  }
+
+  #onClose(event: Event) {
+    event.preventDefault();
```

Comment:
Prevent this method from running again if its already about to close. (when the timer is running)

I guess you could just make a check like `if (this.isClosing) return`

Reply:
The case would be someone smashing the esc button. Then its would be best to avoid that starting multiple timeouts.